modules/local/samtools_replaceheader.nf

```diff
@@ -0,0 +1,60 @@
+process SAMTOOLS_REHEADER {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::samtools=1.17"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/samtools:1.17--h00cdaf9_0' :
+        'biocontainers/samtools:1.17--h00cdaf9_0' }"
+
+    input:
+    tuple val(meta), path(file)
+    path(header)
+
+    output:
+    tuple val(meta), path("${prefix}.${suffix}"), optional:true, emit: bam
+    tuple val(meta), path("${prefix}.${suffix}"), optional:true, emit: cram
```

Comment:
Otherwise the bam files will be put in the `cram` output channel
```suggestion
    tuple val(meta), path("${prefix}.bam") , optional:true, emit: bam
    tuple val(meta), path("${prefix}.cram"), optional:true, emit: cram
```